### PR TITLE
add support for np.append

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title suggests

### CODE REVIEW

File: numba/targets/arrayobj.py [hunk 1]
```diff
@@ -1667,6 +1667,23 @@ def np_reshape_impl(a, shape):
     return np_reshape_impl
 
 
+def _is_nonelike(ty):
+    return (ty is None) or isinstance(ty, types.NoneType)
```
Comment (guilhermeleobas) [inline numba/targets/arrayobj.py]: This function is also defined on arraymath.py. Maybe both definitions should be moved to a file common to both.

https://github.com/numba/numba/blob/60d2bddd0a9ddf84cf900a70e4b7e17bcbe5dc87/numba/targets/arraymath.py#L3665-L3666

Comment (stuartarchibald) [reply numba/targets/arrayobj.py]: Perhaps move it to `numba.numpy_support` for now? There's a load of utility functions there already.

**Comment by guilhermeleobas** [comment]
Done! Ready for another round of review.

File: numba/numpy_support.py [hunk 2]
```diff
@@ -602,3 +602,8 @@ def type_can_asarray(arr):
     """
     ok = (types.Array, types.Sequence, types.Tuple, types.Number, types.Boolean)
     return isinstance(arr, ok)
+
+
+def _is_nonelike(ty):
```
Comment (stuartarchibald) [inline numba/numpy_support.py]: This should be a public function.

File: numba/tests/test_np_functions.py [hunk 3]
```diff
@@ -375,6 +379,21 @@ def check(x_types, x_values):
         x_types = [types.complex64, types.complex128]
         check(x_types, x_values)
 
+    def test_np_append(self):
+        def arrays():
+            yield np.arange(10), 3, None
+            yield np.arange(10), np.arange(3), None
+            yield np.arange(10).reshape(5, 2), np.arange(3), None
+            yield np.array([[1, 2, 3], [4, 5, 6]]), np.array([[7, 8, 9]]), 0
```
Comment (stuartarchibald) [inline numba/tests/test_np_functions.py]: Perhaps add in an `axis = 1` variant?

File: numba/targets/arrayobj.py [hunk 4]
```diff
@@ -1667,6 +1668,19 @@ def np_reshape_impl(a, shape):
     return np_reshape_impl
 
 
+@overload(np.append)
+def np_append(arr, values, axis=None):
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: This needs to make sure that `arr` and `values` are array-like

File: numba/targets/arrayobj.py [hunk 5]
```diff
@@ -1667,6 +1668,19 @@ def np_reshape_impl(a, shape):
...
+    if _is_nonelike(axis):
+        def impl(arr, values, axis=None):
+            arr = np.ravel(np.asarray(arr))
+            values = np.ravel(np.asarray(values))
+            return np.concatenate((arr, values))
+    else:
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: This needs to make sure axis is an `Integer` type

**Comment by stuartarchibald** [comment]
Please could you merge master into this branch to resolve conflicts, thanks.

**Comment by guilhermeleobas** [comment]
I will wait a couple more days to merge master again.

**Comment by esc** [comment]
@guilhermeleobas I don't think it was a specific commit, but that some of the conda packages caused the strange errors. I have just restarted the Azure builds, let's see if it goes anywhere.

**Pull request merged**